libyul/ObjectParser.cpp

```diff
@@ -104,9 +111,62 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::SourceNameMap> ObjectParser::tryParseSourceNameMapping() const
+{
+	// @use-src 0:"abc.sol", 1:"foo.sol", 2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		"(^|\\s)@use-src(\\s+((.|\n)*))?$",
```

Comment:
Documentation: `.` matches `any character except line terminators (LF, CR, LS, PS).`

So `\n` is not enough. You can use `x|[^x]`.

Also https://www.cplusplus.com/reference/regex/ECMAScript/ says that `$` matches the end of string or right before a newline, so this does not work either.

Maybe it is time to switch to just using the scanner?